core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -293,8 +282,8 @@ public synchronized IndexShard createShard(int sShardId, ShardRouting routing) {
                 indexShard = new IndexShard(shardId, indexSettings, path, store, indexServicesProvider);
             }
 
-            indicesLifecycle.indexShardStateChanged(indexShard, null, "shard created");
-            indicesLifecycle.afterIndexShardCreated(indexShard);
+            eventListener.indexShardStateChanged(indexShard, indexShard.state(), null, "shard created");
```

Comment:
This should be `eventListener.indexShardStateChanged(indexShard, null, indexShard.state(), "shard created");